backend/core/src/shared/query-filter/custom_filters.ts

```diff
@@ -5,46 +5,6 @@ import {
 } from "../../listings/types/listing-filter-keys-enum"
 import { filterTypeToFieldMap } from "../../listings/dto/filter-type-to-field-map"
 
-export function addSeniorHousingQuery(qb: WhereExpression, filterValue: string) {
-  const whereParameterName = ListingFilterKeys.seniorHousing
-  const seniorHousingCommunityType = "senior62"
-  const reservedCommunityTypeColumnName = `LOWER(CAST(${
-    filterTypeToFieldMap[ListingFilterKeys.seniorHousing]
-  } as text))`
-  if (filterValue == "true") {
-    qb.andWhere(`${reservedCommunityTypeColumnName} = LOWER(:${whereParameterName})`, {
-      [whereParameterName]: seniorHousingCommunityType,
-    })
-  } else if (filterValue == "false") {
-    qb.andWhere(
-      `(${reservedCommunityTypeColumnName} IS NULL OR ${reservedCommunityTypeColumnName} <> LOWER(:${whereParameterName}))`,
-      {
-        [whereParameterName]: seniorHousingCommunityType,
-      }
-    )
-  }
-}
-
-export function addIndependentLivingHousingQuery(qb: WhereExpression, filterValue: string) {
-  const whereParameterName = ListingFilterKeys.independentLivingHousing
-  const independentLivingCommunityType = "specialNeeds"
-  const reservedCommunityTypeColumnName = `LOWER(CAST(${
-    filterTypeToFieldMap[ListingFilterKeys.independentLivingHousing]
-  } as text))`
-  if (filterValue == "true") {
-    qb.andWhere(`(${reservedCommunityTypeColumnName} = LOWER(:${whereParameterName}))`, {
-      [whereParameterName]: independentLivingCommunityType,
-    })
-  } else if (filterValue == "false") {
-    qb.andWhere(
-      `(${reservedCommunityTypeColumnName} IS NULL OR ${reservedCommunityTypeColumnName} <> LOWER(:${whereParameterName}))`,
-      {
-        [whereParameterName]: independentLivingCommunityType,
-      }
-    )
-  }
-}
-
 export function addAvailabilityQuery(
```

Comment:
I could be missing this but I think we need another query here for bedroom type based on the updates made for unit groups - something like, "does this bedroom type existing in any of the unit groups on this listing"

Reply:
I confirmed how it works right now when you provide following filters:

```
filter[0][$comparison]: =
filter[0][bedrooms]: 3
```

it returns a listing but attached are only those unit groups with bedrooms number equal to 3, which is not what is expected from this filter unfortunately. I need to do some more work.